spec/models/application_spec.rb

```diff
@@ -2,4 +2,7 @@
 
 RSpec.describe Application, type: :model do
   it { should belong_to :user }
+  it { should validate_presence_of(:name).with_message(/can't be blank!/) }
+  it { should validate_presence_of(:url).with_message(/can't be blank!/) }
+  it { should validate_uniqueness_of(:url).with_message(/already belongs to another application/) }
```

Comment:
Consider adding a test for validating uniqueness of :name `scoped_to` `:user_id`. See `scoped_to` on this shoulda-matchers API page: http://www.rubydoc.info/github/thoughtbot/shoulda-matchers/Shoulda%2FMatchers%2FActiveRecord%3Avalidate_uniqueness_of